Library/String+SimpleHTML.swift

```diff
@@ -93,19 +104,29 @@ public extension String {
    - returns: A string with all HTML stripped.
    */
   func htmlStripped(trimWhitespace: Bool = true) -> String? {
-    guard let data = self.data(using: String.Encoding.utf8) else { return nil }
+    func parsedHtml() -> String? {
+      guard let data = self.data(using: String.Encoding.utf8) else { return nil }
 
-    let options: [NSAttributedString.DocumentReadingOptionKey: Any] = [
-      NSAttributedString.DocumentReadingOptionKey.documentType: NSAttributedString.DocumentType.html,
-      NSAttributedString.DocumentReadingOptionKey.characterEncoding: String.Encoding.utf8.rawValue
-    ]
+      let options: [NSAttributedString.DocumentReadingOptionKey: Any] = [
+        NSAttributedString.DocumentReadingOptionKey.documentType: NSAttributedString.DocumentType.html,
+        NSAttributedString.DocumentReadingOptionKey.characterEncoding: String.Encoding.utf8.rawValue
+      ]
 
-    let string = try? NSAttributedString(data: data, options: options, documentAttributes: nil)
-    let result = string?.string
+      let string = try? NSAttributedString(data: data, options: options, documentAttributes: nil)
+      let result = string?.string
 
-    if trimWhitespace {
-      return result?.trimmingCharacters(in: .whitespacesAndNewlines)
+      if trimWhitespace {
+        return result?.trimmingCharacters(in: .whitespacesAndNewlines)
+      }
+      return result
+    }
+
+    if Thread.isMainThread {
+      return parsedHtml()
+    } else {
+      return DispatchQueue.main.sync {
```

Comment:
Same here, should this be `async`?

Reply:
No, `sync` is correct as per other answer.